src/test/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactoryTest.java

```diff
@@ -71,40 +73,95 @@ public void toMapShouldReturnMapWithCorrectCsvDataForPaymentTransaction() {
 
-        TransactionFixture refundTransactionFixture = aTransactionFixture()
-                .withCreatedDate(ZonedDateTime.parse("2018-03-12T16:25:01.123456Z"))
-                .withTransactionType("REFUND")
-                .withAmount(99L)
-                .withTotalAmount(99L)
+        var transactionEntity = transactionFixture
+                .withTransactionType(TransactionType.REFUND.name())
                 .withState(TransactionState.ERROR)
+                .withTotalAmount(123L)
+                .withRefundedByUserEmail("refundedbyuser@example.org")
                 .withGatewayTransactionId("refund-gateway-transaction-id")
                 .withRefundedByUserEmail("refundedbyuser@example.org")
                 .withParentExternalId("parent-external-id")
+                .withCorporateCardSurcharge(0)
                 .withDefaultPaymentDetails()
-                .withDefaultTransactionDetails();
+                .withDefaultTransactionDetails()
+                .toEntity();
 
-        Map<String, Object> csvDataMap = csvTransactionFactory.toMap(refundTransactionFixture.toEntity());
+        Map<String, Object> csvDataMap = csvTransactionFactory.toMap(transactionEntity);
 
-        assertPaymentDetails(csvDataMap, refundTransactionFixture.toEntity());
-        assertThat(csvDataMap.get("Amount"), is("-0.99"));
-        assertThat(csvDataMap.get("Provider ID"), is(refundTransactionFixture.getGatewayTransactionId()));
-        assertThat(csvDataMap.get("GOV.UK Payment ID"), is(refundTransactionFixture.getParentExternalId()));
+        assertPaymentDetails(csvDataMap, transactionEntity);
+        assertThat(csvDataMap.get("Amount"), is("-1.00"));
+        assertThat(csvDataMap.get("Provider ID"), is(transactionFixture.getGatewayTransactionId()));
+        assertThat(csvDataMap.get("GOV.UK Payment ID"), is(transactionFixture.getParentExternalId()));
         assertThat(csvDataMap.get("State"), is("Refund error"));
         assertThat(csvDataMap.get("Finished"), is(true));
         assertThat(csvDataMap.get("Error Code"), is("P0050"));
         assertThat(csvDataMap.get("Error Message"), is("Payment provider returned an error"));
         assertThat(csvDataMap.get("Date Created"), is("12 Mar 2018"));
         assertThat(csvDataMap.get("Time Created"), is("16:25:01"));
         assertThat(csvDataMap.get("Corporate Card Surcharge"), is("0.00"));
-        assertThat(csvDataMap.get("Total Amount"), is("-0.99"));
-        assertThat(csvDataMap.get("Net"), is("-0.99"));
+        assertThat(csvDataMap.get("Total Amount"), is("-1.23"));
+        assertThat(csvDataMap.get("Net"), is("-1.23"));
         assertThat(csvDataMap.get("MOTO"), is(nullValue()));
     }
 
     @Test
-    public void toMapShouldIncludeExternalMetadataFields() {
+    void toMapShouldIncludeNetForFailedTransactionWhenAvailable() {
+
+        TransactionEntity transactionEntity = transactionFixture.withNetAmount(-5L).toEntity();
```

Comment:
Can we set a failed state explicitly ? Default is SUBMITTED and changing it would fail this test.

Reply:
And another test where `Net` is set to `0` if the transaction state is other than SUCCESS ?

Reply:
> And another test where `Net` is set to `0` if the transaction state is other than SUCCESS ?

There is a test that covers this scenario already, the default fixture is in a failure state `toMapShouldReturnMapWithCorrectCsvDataForPaymentTransaction`.